mautic/mautic
# DateTimeHelper use local timezone from configuration not from server

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This is pain from all versions Mautic. DateTimeHelper used date_default_timezone_get() to get local timezone. 
But customers can set default timezone in configuration. We should use as timezone  from configuration, it make more sense. 

Then we do not need take default_timezone from parameters on every place like here https://github.com/mautic/mautic/blob/661b0f07d1c588e01d67901b99f16e715fc72fec/app/bundles/CoreBundle/Helper/DateTime/DateTimeToken.php#L46 

Instead of

```
        $contactTimezone       = $contactTimezone ?: $this->coreParametersHelper->get('default_timezone', 'UTC');
        $dateTime              = new \DateTime('now', new \DateTimeZone($contactTimezone));
```

We can use 

`$dateTime = (new DateTimeHelper())->getLocalDateTime();`



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Probably check If date time is correct on timeline of contact etc
